src/Task/CheckFilePermissions/CheckFilePermissionsTask.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Wunderio\GrumPHP\Task\CheckFilePermissions;
+
+use GrumPHP\Collection\FilesCollection;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use GrumPHP\Task\AbstractExternalTask;
+use Wunderio\GrumPHP\Task\ContextFileExternalTaskBase;
+
+/**
+ * Class CheckFilePermissionsTask.
+ *
+ * @package Wunderio\GrumPHP\CheckFilePermissions
+ */
+class CheckFilePermissionsTask extends ContextFileExternalTaskBase
+{
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getName(): string
+  {
+    return 'check_file_permissions';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigurableOptions(): OptionsResolver
+  {
+    $resolver = new OptionsResolver();
+    $resolver->setDefaults([
+      'ignore_patterns' => ['*/vendor/*','*/node_modules/*', '*/core/*', '*/modules/contrib/*', '*/themes/contrib/*'],
+      'extensions' => ['sh', 'py'],
+      'run_on' => ['.']
+    ]);
+    $resolver->addAllowedTypes('ignore_patterns', ['array']);
+    $resolver->setAllowedTypes('extensions', 'array');
+    $resolver->setAllowedTypes('run_on', ['array']);
+    return $resolver;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function canRunInContext(ContextInterface $context): bool
+  {
+    return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function run(ContextInterface $context): TaskResultInterface
+  {
+    $config = $this->getConfiguration();
+    $files = $this->getFiles($context, $config);
+
+    if (0 === \count($files)) {
+      return TaskResult::createSkipped($this, $context);
+    }
+
+    $arguments = $this->processBuilder->createArgumentsForCommand('check_perms');
+    if (!$context instanceof GitPreCommitContext) {
```

Comment:
What if this whole "if" block is refactored to its own `getFiles` method?

So it just becomes:

```
$files = $this->getFiles();
$arguments->addFiles($files);
```

Reply:
Refactored everything :) Now you need to test again.